radixdlt-engine/src/main/java/com/radixdlt/atommodel/system/state/ValidatorStakeData.java

```diff
@@ -30,22 +30,26 @@
 import com.radixdlt.utils.UInt384;
 
 import java.util.Objects;
+import java.util.OptionalInt;
 
-public final class ValidatorStake implements ResourceInBucket {
+public final class ValidatorStakeData implements ResourceInBucket {
 	public static final UInt256 MINIMUM_STAKE = TokenDefinitionUtils.SUB_UNITS.multiply(UInt256.TEN);
 	public static final int EPOCHS_LOCKED = 1; // Must go through one full epoch before being unlocked
 
 	private final UInt256 totalStake;
 
 	private final UInt256 totalOwnership;
 
+	private final OptionalInt rakePercentage;
```

Comment:
Probably it is also worth to update `MysStakeBalance` to collect `rakePercentage`.